lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -0,0 +1,118 @@
+package starknet.provider
+
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import starknet.data.selectorFromName
+import starknet.data.types.*
+import starknet.utils.DevnetClient
+import starknet.provider.gateway.GatewayProvider
+import starknet.provider.rpc.JsonRpcProvider
+
+class ProviderTest {
+    private var contractAddress: Felt? = null
+
+    companion object {
+        @JvmStatic
+        private val devnetClient = DevnetClient()
+
+        @JvmStatic
+        private fun getProviders(): List<Provider> {
+            return listOf(
+                GatewayProvider(
+                    devnetClient.feederGatewayUrl,
+                    devnetClient.gatewayUrl,
+                    StarknetChainId.TESTNET
+                ),
+                JsonRpcProvider(
+                    devnetClient.rpcUrl,
+                    StarknetChainId.TESTNET
+                )
+            )
+        }
+    }
+
+    @BeforeEach
+    fun before() {
+        devnetClient.start()
+        contractAddress = devnetClient.deployContract("providerTest")
+    }
+
+    @AfterEach
+    fun after() {
+        devnetClient.destroy()
+    }
```

Comment:
Do we start a devnet before each test? This will make test run much much longer. I think having a single devnet is ok.